sdk/storage/azure-storage-blob/tests/test_largest_block_blob.py

```diff
@@ -0,0 +1,348 @@
+        payload_dropping_policy = PayloadDroppingPolicy()
+        credential_policy = format_shared_key_credential([storage_account.name, "dummy"], storage_account_key)
+        self._setup(storage_account, storage_account_key, [payload_dropping_policy, credential_policy])
+        blob_name = self._get_blob_reference()
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+        FILE_PATH = 'largest_blob_from_path.temp.{}.dat'.format(str(uuid.uuid4()))
+        with open(FILE_PATH, 'wb') as stream:
+            largeStream = LargeStream(LARGEST_BLOCK_SIZE, 100 * 1024 * 1024)
+            chunk = largeStream.read()
+            while chunk:
+                stream.write(chunk)
+                chunk = largeStream.read()
+
+        # Act
+        with open(FILE_PATH, 'rb') as stream:
+            blob.upload_blob(stream, max_concurrency=2)
+
+        # Assert
+        self._teardown(FILE_PATH)
+        self.assertEqual(payload_dropping_policy.put_block_counter, 1)
+        self.assertEqual(payload_dropping_policy.put_block_sizes[0], LARGEST_BLOCK_SIZE)
+
+    @pytest.mark.skip(reason="This takes really long time")
+    @pytest.mark.live_test_only
+    @GlobalStorageAccountPreparer()
+    def test_create_largest_blob_from_stream_without_network(self, resource_group, location, storage_account, storage_account_key):
+        payload_dropping_policy = PayloadDroppingPolicy()
+        credential_policy = format_shared_key_credential([storage_account.name, "dummy"], storage_account_key)
+        self._setup(storage_account, storage_account_key, [payload_dropping_policy, credential_policy])
+        blob_name = self._get_blob_reference()
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+
+        number_of_blocks = 50000
+
+        stream = LargeStream(LARGEST_BLOCK_SIZE*number_of_blocks)
+
+        # Act
+        blob.upload_blob(stream, max_concurrency=1)
+
+        # Assert
+        self.assertEqual(payload_dropping_policy.put_block_counter, number_of_blocks)
+        self.assertEqual(payload_dropping_policy.put_block_sizes[0], LARGEST_BLOCK_SIZE)
+
+    @pytest.mark.live_test_only
+    @GlobalStorageAccountPreparer()
+    def test_create_largest_blob_from_stream_single_upload_without_network(self, resource_group, location, storage_account, storage_account_key):
+        payload_dropping_policy = PayloadDroppingPolicy()
+        credential_policy = format_shared_key_credential([storage_account.name, "dummy"], storage_account_key)
+        self._setup(storage_account, storage_account_key, [payload_dropping_policy, credential_policy],
+                    max_single_put_size=LARGEST_SINGLE_UPLOAD_SIZE)
+        blob_name = self._get_blob_reference()
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+
+        stream = LargeStream(LARGEST_SINGLE_UPLOAD_SIZE)
+
+        # Act
+        blob.upload_blob(stream, length=LARGEST_SINGLE_UPLOAD_SIZE, max_concurrency=1)
```

Comment:
>LARGEST_SINGLE_UPLOAD_SIZE [](start = 40, length = 26)

Just curious, I assumed the max single upload size service can accept is 4000Mb, if we send real traffic for this test(I know you are mocking request and response in this test), will service throw error?

Reply:
Max single upload limit is bumped to 5000MB on server side.